### PR TITLE
Upgrade WixToolset to v4.0.1

### PR DESCRIPTION
**Pull request by BMurri**
#### Details

Upgrade WixToolset to v4.0.1

##### Motivation

Address issue #946

##### Context

In WiX v4, it's easier than ever to build MSIs for each platform (x86, x64, arm64) using the exact same sources (with no conditional code).

The default build output directory for the MSI changes from `bin\$(Configuration)\...` to `bin\$(Platform)\$(Configuration)\...` where Platform = `x86`

This MSI contains suboptimal authoring, especially concerning the Windows Installer Best Practices relating to Component/File relationships. Generally speaking, each Component should have only one File. You can create ComponentGroups to group components as needed, but ComponentGroups don't survive in the MSI (they are an authoring-only concept).

#### Pull request checklist

- [x] Addresses an existing issue: #946

### CODE REVIEW

**Comment by DaveTryon** [comment]
I'll need to create a test a signed build before approving this PR

File: docs/BuildingTheCode.md [hunk 1]
```diff
@@ -17,10 +17,13 @@
   cd axe-windows
   ```
 
-### 2. Open the solution in Visual Studio
+### 2. Install the HeatWave extension for Visual Studio
+The `CLI_Installer` project uses WiX 4.X. It will build correctly from the command line, but if you intend to use Visual Studio's IDE, you should consider installing the [HeatWave extension for VS2022](https://marketplace.visualstudio.com/items?itemName=FireGiant.FireGiantHeatWaveDev17). The IDE will continue to work without this extension, but you may encounter errors from the `CLI_Installer` project.
+
+### 3. Open the solution in Visual Studio
 - Use the `src/AxeWindows.sln` file to open the solution.
```
Comment (codeofdusk) [inline docs/BuildingTheCode.md:24]: Why was this line removed?

**Review by codeofdusk — APPROVED** [review]

**Comment by DaveTryon** [comment]
Thanks, @BMurri! Merging this now. The per-user vs per-machine stuff is intentional, even though it's non-ideal.

File: docs/BuildingTheCode.md (hunk 1 above)
Comment (BMurri) [reply docs/BuildingTheCode.md:24]: Probably accidental mousing. The line with the path to the solution file should be restored
